src/lex.cpp

```diff
@@ -1337,6 +1339,75 @@ void lexer::parse_number() {
     QLJS_SLOW_ASSERT(
         !(number_begin[0] == u8'0' && this->is_digit(number_begin[1])));
   }
+  if (!has_decimal_point && !has_exponent && !is_bigint) {
+    std::string cleaned_string = "";
+    for (char8 c : string8_view(number_begin, input - number_begin)) {
+      if (c != '_') {
+        cleaned_string.push_back(static_cast<char>(c));
+      }
+    }
+    if (cleaned_string.size() > 15) {
```

Comment:
Performance: Let's check the length of the string *before* making `cleaned_string`. This avoids copying in the common case (small integer literals).
```suggestion
  string8_view number_literal(number_begin, input - number_begin)
  if (number_literal.size() > 15) {
    std::string cleaned_string = "";
    for (char8 c : number_literal) {
      if (c != '_') {
        cleaned_string.push_back(static_cast<char>(c));
      }
    }
    if (cleaned_string.size() > 15) {
```